packages/docs/content/docs/getting-started/build-your-first-api.mdx

```diff
@@ -0,0 +1,851 @@
+      }
+    }
+    ```
+  </Tab>
+  <Tab value="JS">
+    ```javascript
+    const { z } = require('zod')
+
+    exports.config = {
+      type: 'api',
+      name: 'CreatePet',
+      description: 'Create a new pet in the store',
+      method: 'POST',
+      path: '/pets',
+      
+      bodySchema: z.object({
+        name: z.string(),
+        breed: z.string(),
+        age: z.number(),
+      }),
+
+      responseSchema: {
+        200: z.object({
+          id: z.string(),
+          name: z.string(),
+          breed: z.string(),
+          age: z.number(),
+        }),
+      },
+    }
+
+    exports.handler = async (req, { logger }) => {
+      logger.info('Creating new pet', { body: req.body })
+
+      // In a real app, you will save to a database
+      const newPet = {
+        id: Date.now().toString(),
+        ...req.body,
+      }
+
+      return {
+        status: 200,
+        body: newPet,
+      }
+    }
+    ```
+  </Tab>
+  <Tab value="Python">
+    ```python
+    from pydantic import BaseModel
+    import time
+
+    class PetRequest(BaseModel):
+        name: str
+        breed: str
+        age: int
+
+    class PetResponse(BaseModel):
+        id: str
+        name: str
+        breed: str
+        age: int
+
+    config = {
+        'type': 'api',
+        'name': 'CreatePet',
+        'description': 'Create a new pet in the store',
+        'method': 'POST',
+        'path': '/pets',
+        'bodySchema': PetRequest.model_json_schema(),
+        'responseSchema': {
+            '200': PetResponse.model_json_schema()
+        }
+    }
+
+    async def handler(req, context):
+        context.logger.info('Creating new pet', {'body': req.get('body')})
+        
+        # In a real app, you will save to a database
+        new_pet = {
+            'id': str(int(time.time() * 1000)),
+            **req.get('body')
+        }
+        
+        return {
+            'status': 200,
+            'body': new_pet
+        }
+    ```
+  </Tab>
+</Tabs>
+
+The most fundamental part of any API endpoint is defining what HTTP method it accepts and where it lives:
+
+```typescript
+method: 'POST',
+path: '/pets',
+```
+This configuration tells Motia to create a POST endpoint at /pets. When your development server is running, users can send requests to http://localhost:3000/pets to create new pets. The POST method is the standard HTTP method for creating new resources in REST APIs.
+
+### Request Validation with Zod
+The bodySchema is where we define what data our endpoint expects and how it should be validated:
+```typescript
+bodySchema: z.object({
+  name: z.string(),
+  breed: z.string(),
+  age: z.number(),
+}),
+```
+This schema serves two important purposes. First, it provides compile-time TypeScript types throughout your handler function. When you access `req.body.name`, TypeScript knows it's a string and can provide proper autocomplete and error checking.
+
+![Pet Creation](./../img/pet-creation.png)
```

Comment:
we don't have built-in validation (we don't want to be an opinionated framework, we want to give developers freedom to implement their own validation layer)